<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="generator" content="Hugo 0.44" />


<title>One way to evaluate incomplete lineage sorting in your phylogenetic or phylogeographic data - blog</title>
<meta property="og:title" content="One way to evaluate incomplete lineage sorting in your phylogenetic or phylogeographic data - blog">



  







<link rel="stylesheet" href="/css/fonts.css" media="all">
<link rel="stylesheet" href="/css/main.css" media="all">



  </head>
  <body>
    <div class="wrapper">
      <header class="header">
        <nav class="nav">
  <a href="/" class="nav-logo">
    <img src="/images/jselfie_logo.png"
         width="50"
         height="50"
         alt="Logo">
  </a>

  <ul class="nav-links">
    
    <li><a href="/about/">About</a></li>
    
    <li><a href="https://github.com/justincbagley">GitHub</a></li>
    
    <li><a href="https://www.linkedin.com/in/justin-c-bagley-2a8a0523">LinkedIn</a></li>
    
    <li><a href="https://www.researchgate.net/profile/Justin_Bagley2">ResearchGate</a></li>
    
    <li><a href="https://twitter.com/justincbagley">Twitter</a></li>
    
  </ul>
</nav>

      </header>


<main class="content" role="main">

  <article class="article">
    
    <span class="article-duration">2 min read</span>
    

    <h1 class="article-title">One way to evaluate incomplete lineage sorting in your phylogenetic or phylogeographic data</h1>

    
    <span class="article-date">2011/07/19</span>
    

    <div class="article-content">
      <p>There aren&rsquo;t many ways to quantitatively evaluate incomplete lineage sorting (ILS) or ancestral polymorphism in molecular data in a statistically rigorous way.  One means of gauging relative amounts of lineage sorting is the gsi, which I will be working on later.  But, right here I wanted to note that I recently saw Ian Wang &amp; Brad Shaffer using Wayne Maddison and Lacey Knowles method to evaluate ILS in a paper they wrote on dendrobatid frogs of Costa Rica, which is one of the main areas I am studying in my dissertation research.  Here is an excerpt from their paper:</p>

<blockquote>
<p>&ldquo;Because incomplete lineage sorting has the potential to confound<br />
phylogenetic analyses of recently diverged populations, we<br />
also performed the “minimize deep coalescences method” following<br />
the procedure of Maddison and Knowles (2006). This method<br />
takes into account the process of lineage sorting by genetic drift<br />
and information on the genealogical relationships among alleles<br />
(Maddison and Knowles 2006) and has been shown to improve the<br />
probability of recovering an accurate population tree, even with a<br />
single locus (Knowles and Carstens 2007). We first performed a<br />
parsimony analysis in PAUP∗, using a heuristic search with 1000<br />
random addition sequence replicates and maxtrees = 100, and<br />
saved a majority-rule consensus tree from all equally parsimonious<br />
trees recovered in the search. We then used the tree search<br />
tool in Mesquite version 2.5 (Maddison and Maddison 2005) to<br />
find the population tree that minimized the total number of deep<br />
coalescences, using the saved majority-rule consensus tree and<br />
SPR branch swapping (Maddison and Knowles 2006).&rdquo;</p>
</blockquote>

<p>Through this analysis, the authors were able to say that their phylogeny based on their mtDNA data was not different from a tree where ILS was minimized, suggesting the tree more likely reflected historical signal. </p>

<p>This is a nifty little trick that I plan on using.  I am currently learning to implement gene tree modeling in MESQUITE, but I have a long way to go (feels like)&hellip;.  my consolation is that I am already really familiar with the basics and with phylogenetic comparative analyses and trait evolution-related analyses in the program, so maybe my progress will be fast. </p>

<p><strong>References</strong></p>

<p>Ian J. Wang1,2 and H. Bradley Shaffer1 (2008) RAPID COLOR EVOLUTION IN AN APOSEMATIC SPECIES: A PHYLOGENETIC ANALYSIS OF COLOR VARIATION IN THE STRIKINGLY POLYMORPHIC STRAWBERRY POISON-DART FROG. <em>Evolution</em>, 62-11: 2742–2759.</p>

    </div>
  </article>

  


</main>

      <footer class="footer">
        <ul class="footer-links">
          <li>
            <a href="/index.xml" type="application/rss+xml" target="_blank">RSS feed</a>
          </li>
          <li>
            <a href="https://gohugo.io/" class="footer-links-kudos">Made with <img src="/images/hugo-logo.png" width="22" height="22"></a>
          </li>
        <p class="m-0 text-center text-white">Copyright &copy; 2018 Justin C. Bagley</p>
        </ul>
      </footer>

    </div>
    

    

    
  </body>
</html>

