---
author: justin
comments: true
date: 2017-09-13 11:40:35+00:00
layout: post
link: http://www.justinbagley.org/3160/gist-ddradseq-data-snapp
slug: gist-ddradseq-data-snapp
title: Analyzing ddRADseq Data in SNAPP
wordpress_id: 3160
categories:
- blog posts
- ddRAD-seq
- Gist
- GitHub
- Rad-seq
- shell scripts
- SNAPP
- SNPs
- species trees
tags:
- alignment
- BEAST
- BEAST2
- BEAUti
- ddRAD-seq
- fastq
- FastQC
- Gist
- GitHub
- NEXUS
- phrynomics
- pyRAD
- R
- shell script
- SNAPP
- SNPs
---

[![](http://www.justinbagley.org/wp-content/uploads/2017/09/GitHub_Gist_img.png)](https://gist.github.com)Oh, SNAPP...



Over the last year, as I've become increasingly comfortable with [GitHub](https://github.com), I've naturally shifted all of my note writing to [Markdown](https://daringfireball.net/projects/markdown/) language. During the process, I've started frequently jotting down useful notes related to analysis, code, or scientific computing as "Gists" on GitHub. In contrast to most repositories, Gists are small snippets of code or notes and are posted at [gist.github.com](https://gist.github.com) rather than [github.com](https://github.com). To date, I have published 8 Gists on topics as broad as Pandoc, ∂a∂i, SNAPP, and Markdown.

**Recently, I published an updated public version of an earlier [Gist](https://gist.github.com/justincbagley/86b8c28280abc2a71e90bbd652dc9eb1) containing a broad description of steps for analyzing SNP (single nucleotide polymorphism) data from ddRADseq loci in the software program [SNAPP](https://github.com/BEAST2-Dev/SNAPP) (Bryant et al. 2012; implemented in [BEAST](https://www.beast2.org) v2.4++), to infer species trees with divergence times.** The new Gist lists 8 steps for doing this, which take advantage of several available software programs, as well as my own shell scripts. In the spirit of open science, my motivation for writing this note was that, while people don't necessarily need another full-blown, in-depth tutorial on this (e.g. they can read [this one](http://evomicsorg.wpengine.netdna-cdn.com/wp-content/uploads/2016/01/BFDstar-tutorial1.pdf)), I could provide a set of steps that _I used_ to prepare and analyze SNP data in SNAPP for recent projects. 

Here is a link to the [Gist](https://gist.github.com/justincbagley/8f1a263790609ed5275db065aa96b42b).

Here, I am sharing this Gist with you as embedded Markdown, thanks to the [oEmbed Gist](https://github.com/miya0001/oembed-gist) plugin:

https://gist.github.com/justincbagley/8f1a263790609ed5275db065aa96b42b

Cheers, ~ J


