---
author: Justin Bagley
comments: true
date: 2020-12-14 10:49:28+00:00
layout: post
link: http://justinbagley.rbind.io/2020/12/14/new-updates-to-piranha-close-to-new-release/
slug: new-updates-to-piranha-close-to-new-release
title: New updates for PIrANHA, close to new release
categories:
- blog posts
- PIrANHA
- piranha
- software
- shell scripts
tags:
- alignment
- automation
- BEAST
- BEAST2
- BEAUti
- ddRAD-seq
- FASTA
- fastq
- FastQC
- Gist
- GitHub
- NEXUS
- PHYLIP
- phylogenetics
- phylogeography
- PIrANHA
- piranha
- pyRAD
- R
- shell script
- SNPs
- supercomputer
---

<a href="https://imgur.com/AQte6eh"><img src="https://i.imgur.com/AQte6eh.png" title="source: Justin C. Bagley" width=40% height=40% align="center" /></a>

PIrANHA provides a set of tools for automating file processing and analysis steps in the (phylo\*=) fields of phylogenomics and phylogeography (including population genomics). PIrANHA is fully command line-based and contains a series of functions for automating tasks during evolutionary analyses of genetic data. 

A variety of functions manipulate DNA sequence alignments, while others conduct custom analysis pipelines; for example, one set conducts reference-based assembly, allele phasing, and alignment of allelic sequences, starting from cleaned targeted sequence capture reads. Many functions are wrappers around existing software, allowing for straightforward automation of common analysis steps in evolutionary genetics. PIrANHA is under development (join in!), but the [alpha release](https://github.com/justincbagley/piranha/releases) is now stable! 

_New features_ include tab completion of function names, as follows:

<!-- ![piranha-tab-completion](https://raw.githubusercontent.com/justincbagley/piranha/master/assets/piranha_tab_completion2-min.gif) -->
<p align="center"><img src="https://github.com/justincbagley/piranha/tree/master/assets/piranha_tab_completion2-min.gif?raw=true"/></p>

Cheers, ~J


